qiskit/primitives/utils.py

```diff
@@ -68,6 +77,11 @@ def init_observable(observable: BaseOperator | str) -> SparsePauliOp:
         return SparsePauliOp(observable)
 
 
+@deprecate_func(
+    since="1.2",
+    additional_msg="Use ``mthree.utils.final_measurement_mapping`` instead. "
+    + "See https://qiskit-extensions.github.io/mthree/apidocs/utils.html for details.",
```

Comment:
A better suggestion here might be to point people to `QuantumCircuit.layout` and `SparsePauliOp.apply_layout` to adjust an operator for a layout instead of tracking permutations with measurements. Although also pointing to `mthree` if they want the same exact functionality does make sense too.

Reply:
Thank you for the suggestion. I will add `layout` and `apply_layout` in the message as well as the existing mthree message.